cicerone/spec/parameter.py

```diff
@@ -0,0 +1,64 @@
+"""Parameter model for OpenAPI parameters.
+
+References:
+- OpenAPI 3.x Parameter Object: https://spec.openapis.org/oas/v3.1.0#parameter-object
+- Swagger 2.0 Parameter Object: https://swagger.io/specification/v2/#parameter-object
+"""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+from pydantic import BaseModel, Field
+
+if TYPE_CHECKING:
+    from cicerone.spec.schema import Schema
+
+
+class Parameter(BaseModel):
+    """Represents an OpenAPI parameter object."""
+
+    # Allow extra fields to support vendor extensions and future spec additions
+    model_config = {"extra": "allow"}
+
+    name: str | None = None
+    in_: str | None = Field(None, alias="in")
+    description: str | None = None
+    required: bool = False
+    schema_: Schema | None = Field(None, alias="schema")
+    # Swagger 2.0 fields
+    type: str | None = None
+    # OpenAPI 3.x fields
+    style: str | None = None
+    explode: bool | None = None
+    example: Any | None = None
+    examples: dict[str, Any] = Field(default_factory=dict)
```

Comment:
Examples is defined as this:

```
Map[ string, [Example Object](https://spec.openapis.org/oas/v3.1.0#example-object) | [Reference Object](https://spec.openapis.org/oas/v3.1.0#reference-object)]
```

We should not use Any - match it to the schema definition.